fake.go

```diff
@@ -519,6 +521,177 @@ func (fake *Fake) Dump() string {
 	return buf.String()
 }
 
+// extractComment removes comment section from a given string and returns it together with comment string.
+// If a comment wasn't found, an empty comment string will be returned.
+func extractComment(s string) (string, string) {
+	if !strings.Contains(s, "comment") {
+		return s, ""
+	}
+	// find comment section with colon
+	reg := regexp.MustCompile(`comment.*".*" ?;?`)
+	// extract comment section and leave just the string
+	comment := reg.FindStringSubmatch(s)[0]
+	commentReg := regexp.MustCompile(`"[^"]+"`)
+	comment = commentReg.FindStringSubmatch(comment)[0]
+	comment = strings.Trim(comment, "\"")
+	// remove comment section from the initial string
+	res := reg.ReplaceAllString(s, "${1}")
+	return strings.TrimSpace(res), comment
+}
+
+// ParseNFTablesDump can parse a dump for a given nft instance.
+// It expects one table declaration and the same nftables family in all rules based on a given nft.nftContext.
+// It lacks some error handling, and the best way to verify that everything important was properly parsed is to
+// compare given data with nft.Dump() output.
+func ParseNFTablesDump(nft *Fake, data string) (err error) {
```

Comment:
Should be a `Fake` method rather than a function.

You don't need `NFTables` in the name... that's already implied by where it is. So just `ParseDump` would be good. (Or `Load`? That's not quite the opposite of `Dump`, but it works...)

Reply:
yeah you can see I started doing it in the kubeproxy file and didn't change the signature :)
I like ParseDump more, as it is more clear on the format it accepts